jobs/src/main/protobuf/metronome.proto

```diff
@@ -78,6 +78,42 @@ message JobRun {
   }
   repeated JobRunTask tasks = 6;
   optional int64 starting_deadline_seconds = 7;
+
+  message JobRunSpecOverrides {
+    repeated EnvironmentVariable environment = 1;
+    optional PlacementSpec placement = 2;
+  }
+  optional JobRunSpecOverrides overrides = 8;
+}
+
+/**
+ * Key, value pair used to store environment variables.
+ */
+message EnvironmentVariable {
+  optional string key = 1;
+  optional string value = 2;
+}
+message PlacementSpec {
```

Comment:
`EnvironmentVariable` and `PlacementSpec` were moved out of `JobSpec` since proto2 doesn't allow referencing nested types.